sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -82,6 +82,53 @@ case class GpuConcat(children: Seq[Expression]) extends GpuComplexTypeMergingExp
   }
 }
 
+case class GpuMapConcat(children: Seq[Expression]) extends GpuComplexTypeMergingExpression {
+
+  @transient override lazy val dataType: MapType = {
+    if (children.isEmpty) {
+      MapType(StringType, StringType)
+    } else {
+      super.dataType.asInstanceOf[MapType]
+    }
+  }
+
+  override def nullable: Boolean = children.exists(_.nullable)
+
+  override def columnarEval(batch: ColumnarBatch): Any = (dataType, children.length) match {
+    // Explicitly return null for empty concat as Spark, since cuDF doesn't support empty concat.
+    case (dt, 0) => GpuColumnVector.fromNull(batch.numRows(), dt)
+    // For single column concat, we pass the result of child node to avoid extra cuDF call.
+    case (_, 1) => children.head.columnarEval(batch)
+    case (dt, _) => {
+      val cols = children.safeMap(columnarEvalToColumn(_, batch))
+      // concatenate keys and values
+      val (key_list, value_list) = withResource(cols) { cols =>
+        withResource(ArrayBuffer[ColumnView]()) { keys => 
+          withResource(ArrayBuffer[ColumnView]()) { values =>
+            cols.foreach{ col =>
+              keys.append(GpuMapUtils.getKeysAsListView(col.getBase))
+              values.append(GpuMapUtils.getValuesAsListView(col.getBase))
+            }    
+            (ColumnVector.listConcatenateByRow(keys: _*), 
+              ColumnVector.listConcatenateByRow(values: _*))
+          }
+        }
+      }
+      // build map column from concatenated keys and values
+      withResource(Seq(key_list, value_list)) { case Seq(keys, values) =>
+        withResource(Seq(keys.getChildColumnView(0), values.getChildColumnView(0))) { 
+          case Seq(k_child, v_chlid) =>
+            withResource(ColumnView.makeStructView(k_child, v_chlid)) {structs =>
+              withResource(keys.replaceListChild(structs)) { struct_list =>
+                GpuCreateMap.createMapFromKeysValuesAsStructs(dt, struct_list)
+              }
+            }
+        }
+      }
+    }
+  }
```

Comment:
Sorry one more thing that actually blocks this from going in. This needs to set hasSideEffects if 
```
(SQLConf.get.getConf(SQLConf.MAP_KEY_DEDUP_POLICY) ==
        SQLConf.MapKeyDedupPolicy.EXCEPTION.toString
```

We need to do it for GpuCreateMap too. That is a bug and I'll file something for it.

Reply:
Yes GpuMapConcat needs
```
 override lazy val hasSideEffects: Boolean =
    GpuCreateMap.exceptionOnDupKeys || super.hasSideEffects
```

Just like I added to GpuCreateMap in #5546

Reply:
Done!